File: ControlRH/wwwroot/js/admin/documento.js
window.Documento = {

    init: function () {
        $(document).ready(function () {
            // dispara quando muda a carteira
            $("#CarteiraClienteId").on("change", function () {
                var carteiraId = $(this).val();

                if (!carteiraId || carteiraId === "00000000-0000-0000-0000-000000000000") {
                    // limpa colaboradores se não tiver carteira válida
                    Documento.limparColaboradores();
                    return;
                }

                Documento.carregarColaboradores(carteiraId);
            });
        });
    },

    carregarColaboradores: function (carteiraId) {
        $.getJSON(`/Admin/Documento/ColaboradoresPorCarteira`, { carteiraClienteId: carteiraId })
            .done(function (data) {
                var $select = $("#ColaboradorId");
                $select.empty();
                $select.append($("<option>").val("").text("-- Selecione --"));

                $.each(data, function (i, colaborador) {
                    $select.append($("<option>").val(colaborador.id).text(colaborador.nome));
                });
            })
            .fail(function () {
                alert("Erro ao carregar colaboradores.");
            });
    },

    limparColaboradores: function () {
        var $select = $("#ColaboradorId");
        $select.empty();
        $select.append($("<option>").val("").text("-- Selecione --"));
    }
};

File: ControlRH/Areas/Admin/Models/Documento.cs
using ControlRH.Core.Models;

namespace ControlRH.Areas.Admin.Models;

public class Documento : AggregateRoot
{
    protected Documento() { }

    public Documento(string nome, string descricao, string caminhoArquivo, Guid carteiraClienteId, Guid? colaboradorId = null)
    {
        Nome = nome;
        Descricao = descricao;
        CaminhoArquivo = caminhoArquivo;
        CarteiraClienteId = carteiraClienteId;
        ColaboradorId = colaboradorId;
    }

    public string Nome { get; private set; }
    public string? Descricao { get; private set; }
    public string CaminhoArquivo { get; private set; }
    public Guid CarteiraClienteId { get; private set; }
    public CarteiraCliente CarteiraCliente { get; private set; }
    public Guid? ColaboradorId { get; private set; }
    public Colaborador? Colaborador { get; private set; }
}
